### ISSUE TITLE
`curatedMetagenomicData("*metaphlan*", dryrun = FALSE)` errors

### ISSUE THREAD

**Issue by lwaldron** [waldronlab/curatedMetagenomicData#162]
`curatedMetagenomicData("*metaphlan*", dryrun = FALSE)` errors at:

```
Working on WenC_2017.metaphlan_bugs_list.stool
Error in as.character(sys.call(sys.parent())[[1L]]) : 
  cannot coerce type 'closure' to vector of type 'character'
```

And `curatedMetagenomicData("WenC_2017.metaphlan_bugs_list.stool", dryrun = FALSE)` returns a strange error:

```
Error: Use Bengtsson-PalmeJ instead of Bengtsson_PalmeJ,
                    Castro-NallarE instead of Castro_NallarE,
                    Heitz-BuschartA instead of Heitz_BuschartA,
                    Obregon-TitoAJ instead of Obregon_TitoAJ,
                    and ChengpingW_2017 instead of WenC_2017
```

(note that the first four of these substitutions are identical). Found in bioc-release, with clear cache.

```
> devtools::session_info()
─ Session info ────────────────────────────────────────────────────────────────────────────────────
 setting  value                       
 version  R version 3.6.3 (2020-02-29)
 os       Ubuntu 18.04.4 LTS          
 system   x86_64, linux-gnu           
 ui       RStudio                     
 language (EN)                        
 collate  en_US.UTF-8                 
 ctype    en_US.UTF-8                 
 tz       America/New_York            
 date     2020-03-10                  

─ Packages ────────────────────────────────────────────────────────────────────────────────────────
 package                * version  date       lib source        
 AnnotationDbi            1.48.0   2019-10-29 [1] Bioconductor  
 AnnotationHub          * 2.18.0   2019-10-29 [1] Bioconductor  
 assertthat               0.2.1    2019-03-21 [1] CRAN (R 3.6.1)
 backports                1.1.5    2019-10-02 [1] CRAN (R 3.6.1)
 Biobase                * 2.46.0   2019-10-29 [1] Bioconductor  
 BiocFileCache          * 1.10.2   2019-11-08 [1] Bioconductor  
 BiocGenerics           * 0.32.0   2019-10-29 [1] Bioconductor  
 BiocManager              1.30.10  2019-11-16 [1] CRAN (R 3.6.1)
 BiocVersion              3.10.1   2019-06-06 [1] Bioconductor  
 bit                      1.1-15.2 2020-02-10 [1] CRAN (R 3.6.2)
 bit64                    0.9-7    2017-05-08 [1] CRAN (R 3.6.1)
 blob                     1.2.1    2020-01-20 [1] CRAN (R 3.6.2)
 callr                    3.4.2    2020-02-12 [1] CRAN (R 3.6.2)
 cli                      2.0.2    2020-02-28 [1] CRAN (R 3.6.3)
 crayon                   1.3.4    2017-09-16 [1] CRAN (R 3.6.1)
 curatedMetagenomicData * 1.16.0   2019-11-05 [1] Bioconductor  
 curl                     4.3      2019-12-02 [1] CRAN (R 3.6.1)
 DBI                      1.1.0    2019-12-15 [1] CRAN (R 3.6.2)
 dbplyr                 * 1.4.2    2019-06-17 [1] CRAN (R 3.6.1)
 desc                     1.2.0    2018-05-01 [1] CRAN (R 3.6.1)
 devtools                 2.2.2    2020-02-17 [1] CRAN (R 3.6.2)
 digest                   0.6.25   2020-02-23 [1] CRAN (R 3.6.2)
 dplyr                  * 0.8.5    2020-03-07 [1] CRAN (R 3.6.3)
 ellipsis                 0.3.0    2019-09-20 [1] CRAN (R 3.6.1)
 ExperimentHub          * 1.12.0   2019-10-29 [1] Bioconductor  
 fansi                    0.4.1    2020-01-08 [1] CRAN (R 3.6.2)
 fastmap                  1.0.1    2019-10-08 [1] CRAN (R 3.6.1)
 fs                       1.3.2    2020-03-05 [1] CRAN (R 3.6.3)
 glue                     1.3.1    2019-03-12 [1] CRAN (R 3.6.1)
 htmltools                0.4.0    2019-10-04 [1] CRAN (R 3.6.1)
 httpuv                   1.5.2    2019-09-11 [1] CRAN (R 3.6.1)
 httr                     1.4.1    2019-08-05 [1] CRAN (R 3.6.1)
 interactiveDisplayBase   1.24.0   2019-10-29 [1] Bioconductor  
 IRanges                  2.20.2   2020-01-13 [1] Bioconductor  
 later                    1.0.0    2019-10-04 [1] CRAN (R 3.6.1)
 lifecycle                0.2.0    2020-03-06 [1] CRAN (R 3.6.3)
 magrittr                 1.5      2014-11-22 [1] CRAN (R 3.6.1)
 memoise                  1.1.0    2017-04-21 [1] CRAN (R 3.6.1)
 mime                     0.9      2020-02-04 [1] CRAN (R 3.6.2)
 pillar                   1.4.3    2019-12-20 [1] CRAN (R 3.6.2)
 pkgbuild                 1.0.6    2019-10-09 [1] CRAN (R 3.6.1)
 pkgconfig                2.0.3    2019-09-22 [1] CRAN (R 3.6.1)
 pkgload                  1.0.2    2018-10-29 [1] CRAN (R 3.6.1)
 prettyunits              1.1.1    2020-01-24 [1] CRAN (R 3.6.2)
 processx                 3.4.2    2020-02-09 [1] CRAN (R 3.6.2)
 promises                 1.1.0    2019-10-04 [1] CRAN (R 3.6.1)
 ps                       1.3.2    2020-02-13 [1] CRAN (R 3.6.2)
 purrr                    0.3.3    2019-10-18 [1] CRAN (R 3.6.1)
 R6                       2.4.1    2019-11-12 [1] CRAN (R 3.6.1)
 rappdirs                 0.3.1    2016-03-28 [1] CRAN (R 3.6.1)
 Rcpp                     1.0.3    2019-11-08 [1] CRAN (R 3.6.1)
 remotes                  2.1.1    2020-02-15 [1] CRAN (R 3.6.2)
 rlang                    0.4.5    2020-03-01 [1] CRAN (R 3.6.3)
 rprojroot                1.3-2    2018-01-03 [1] CRAN (R 3.6.1)
 RSQLite                  2.2.0    2020-01-07 [1] CRAN (R 3.6.2)
 rstudioapi               0.11     2020-02-07 [1] CRAN (R 3.6.2)
 S4Vectors                0.24.3   2020-01-18 [1] Bioconductor  
 sessioninfo              1.1.1    2018-11-05 [1] CRAN (R 3.6.1)
 shiny                    1.4.0    2019-10-10 [1] CRAN (R 3.6.1)
 testthat                 2.3.2    2020-03-02 [1] CRAN (R 3.6.3)
 tibble                   2.1.3    2019-06-06 [1] CRAN (R 3.6.1)
 tidyr                    1.0.2    2020-01-24 [1] CRAN (R 3.6.2)
 tidyselect               1.0.0    2020-01-27 [1] CRAN (R 3.6.2)
 usethis                  1.5.1    2019-07-04 [1] CRAN (R 3.6.1)
 vctrs                    0.2.3    2020-02-20 [1] CRAN (R 3.6.2)
 withr                    2.1.2    2018-03-15 [1] CRAN (R 3.6.1)
 xtable                   1.8-4    2019-04-21 [1] CRAN (R 3.6.1)
 yaml                     2.2.1    2020-02-01 [1] CRAN (R 3.6.2)

[1] /usr/local/lib/R/site-library
[2] /usr/lib/R/site-library
[3] /usr/lib/R/library
```

**Comment by schifferl** [comment]
Strange indeed, I will take a look at this and report back.

**Comment by schifferl** [comment]
It's a simple fix, `WenC_2017.metaphlan_bugs_list.stool` just needs to be added to the exclusion regex in the `curatedMetagenomicData()` function – I will get it done tomorrow and push to BioC.

**Comment by lwaldron** [comment]
It should exclude all the `WenC_2017` data types, not just metaphlan_bugs.

**Comment by schifferl** [comment]
Fixed by e6c1dd7 and 80ca0c9, both have been pushed to release and devel, respectively.
